shared/react-native/ios/Keybase/NativeLogger.m

```diff
@@ -0,0 +1,42 @@
+#import "NativeLogger.h"
+#import <UIKit/UIKit.h>
+#import "AppDelegate.h"
+
+@import CocoaLumberjack;
+@implementation NativeLogger
+
+static const DDLogLevel ddLogLevel = DDLogLevelDebug;
+static const NSString* tagName = @"KBNativeLogger";
+
+RCT_EXPORT_MODULE(KBNativeLogger);
+
+RCT_REMAP_METHOD(log,
+                 tagPrefix:(NSString*)tagPrefix
+                 toLog:(NSString*)toLog)
+{
+  DDLogInfo(@"%@%@: %@", tagPrefix, tagName, toLog);
+}
+
+RCT_REMAP_METHOD(dump,
+                 tagPrefix:(NSString*)tagPrefix
+                 resolver:(RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject)
+{
+  NSMutableArray<NSString *> *lines = [[NSMutableArray alloc] init];
+  AppDelegate *appDel = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+  DDFileLogger *fileLogger = (DDFileLogger *)appDel.fileLogger;
+  NSArray<NSString *> *paths = [[fileLogger logFileManager] sortedLogFilePaths];
+  for (NSString *path in paths) {
+    NSString *fileContents = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:NULL];
+    for (NSString *line in [fileContents componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]]) {
+      NSRange range = [line rangeOfString:[NSString stringWithFormat:@"%@%@: ", tagPrefix, tagName]];
+      if (range.location != NSNotFound) {
+        [lines addObject:[line substringFromIndex:range.location + range.length]];
+      }
+    }
+  }
+  NSArray<NSString *> *finalLines = [lines copy];
```

Comment:
why do we need to do a copy?

Reply:
I'm not sure! the NSMutableArray is an NSArray type, but I was afraid of leaking memory with the mutable array. I don't really understand when the object is released.

Reply:
Looking over the docs at https://facebook.github.io/react-native/docs/native-modules-ios.html , it talks about converting javascript arguments to ObjC, but doesn't say anything about converting ObjC return values to javascript. But I assume that's happening. If we have ARC turned on, then hopefully (?) the array we allocate here gets released properly.

In any case, the same thing would apply to any copy we make, so I don't think the copy is necessary.